### ISSUE TITLE
Unneeded `max` substraction in softmax?

### ISSUE THREAD

**Issue by hypnopump** [lucidrains/PaLM-pytorch#9]
It seems [this operation](https://github.com/lucidrains/PaLM-pytorch/blob/0f76ad22fe15a3a48e7ec98c251feb10c877c6e4/palm_pytorch/palm_pytorch.py#L159) might not be needed after all? [It's done by default](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/SoftMax.cpp#L188)

I can also see the same pattern in: 
* [AF2](https://github.com/lucidrains/alphafold2/blob/main/alphafold2_pytorch/alphafold2.py#L171)
* [NUWA](https://github.com/lucidrains/nuwa-pytorch/blob/main/nuwa_pytorch/nuwa_pytorch.py#L73)
* [DALLE2](https://github.com/lucidrains/DALLE2-pytorch/blob/main/dalle2_pytorch/dalle2_pytorch.py#L690)

Hope this helps!

**Comment by lucidrains** [comment]
@hypnopump great find Eric! looks like i have some spring cleaning to do for all my attention implementations! :pray:
